src/runtime_src/xdp/profile/plugin/aie_trace/CMakeLists.txt

```diff
@@ -50,20 +27,21 @@ file(GLOB AIE_DRIVER_COMMON_UTIL_FILES
   "${PROFILE_DIR}/device/common/*.cpp"
 )
 
-if (XDP_VE2_BUILD_CMAKE STREQUAL "yes")
-  add_library(xdp_aie_trace_plugin MODULE ${AIE_TRACE_PLUGIN_FILES} ${AIE_TRACE_UTIL_FILES} ${AIE_TRACE_CONFIG_FILES})
-  add_dependencies(xdp_aie_trace_plugin xdp_core xrt_coreutil)
-  target_link_libraries(xdp_aie_trace_plugin PRIVATE xdp_core xrt_coreutil xaiengine)
-  target_compile_definitions(xdp_aie_trace_plugin PRIVATE XDP_VE2_BUILD=1 FAL_LINUX="on")
-  target_include_directories(xdp_aie_trace_plugin PRIVATE ${CMAKE_SOURCE_DIR}/src)
-  set_target_properties(xdp_aie_trace_plugin PROPERTIES VERSION ${XRT_VERSION_STRING} SOVERSION ${XRT_SOVERSION})
+if (XDP_CLIENT_BUILD_CMAKE STREQUAL "yes")
+  set(IMPL_DIR "${PROFILE_DIR}/plugin/aie_trace/client")
+  set(OFFLOAD_DIR "${PROFILE_DIR}/device/aie_trace/client")
+  set(DEVICE_DIR "${PROFILE_DIR}/device/client_device")
 
-  install (TARGETS xdp_aie_trace_plugin
-    LIBRARY DESTINATION ${XDP_PLUGIN_INSTALL_DIR} COMPONENT ${XRT_COMPONENT}
+  file(GLOB AIE_TRACE_COMPONENT_FILES
+    "${OFFLOAD_DIR}/*.h"
+    "${OFFLOAD_DIR}/*.cpp"
+    "${DEVICE_DIR}/*.h"
+    "${DEVICE_DIR}/*.cpp"
+    "${IMPL_DIR}/*.h"
+    "${IMPL_DIR}/*.cpp"
   )
 
-elseif (XDP_CLIENT_BUILD_CMAKE STREQUAL "yes")
-  add_library(xdp_aie_trace_plugin MODULE ${AIE_TRACE_PLUGIN_FILES} ${AIE_TRACE_UTIL_FILES} ${AIE_DRIVER_COMMON_UTIL_FILES})
+  add_library(xdp_aie_trace_plugin MODULE ${AIE_TRACE_PLUGIN_FILES} ${AIE_TRACE_COMPONENT_FILES} ${AIE_TRACE_UTIL_FILES} ${AIE_DRIVER_COMMON_UTIL_FILES})
```

Comment:
AIE_DRIVER_COMMON_UTIL_FILES is only used in Client Build. I would suggest to move it under XDP_CLIENT_BUILD_CMAKE.